tests/dead_letter_queue/test_dead_letter_queue.py

```diff
@@ -0,0 +1,189 @@
+import time
+from datetime import datetime
+from typing import Callable, Mapping, MutableSequence, Optional, Tuple
+
+import pytest
+
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.dead_letter_queue.dead_letter_queue import DeadLetterQueue
+from arroyo.dead_letter_queue.factory import DeadLetterQueueFactory
+from arroyo.dead_letter_queue.policies.abstract import InvalidMessage
+from arroyo.dead_letter_queue.policies.count import CountInvalidMessagePolicy
+from arroyo.dead_letter_queue.policies.ignore import IgnoreInvalidMessagePolicy
+from arroyo.dead_letter_queue.policies.raise_e import RaiseInvalidMessagePolicy
+from arroyo.processing.strategies.abstract import (
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
+from arroyo.types import Message, Partition, Position, Topic
+
+
+class FakeProcessingStep(ProcessingStrategy[KafkaPayload]):
+    """
+    Raises InvalidMessage if a submitted message has no key in payload.
+    """
+
+    def poll(self) -> None:
+        pass
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        pass
+
+    def terminate(self) -> None:
+        pass
+
+    def close(self) -> None:
+        pass
+
+    def submit(self, message: Message[KafkaPayload]) -> None:
+        """
+        Valid message is one with a key.
+        """
+        if message.payload.key is None:
+            raise InvalidMessage
+
+
+class FakeProcessingStepFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def create(
+        self, commit: Callable[[Mapping[Partition, Position]], None]
+    ) -> ProcessingStrategy[KafkaPayload]:
+        return FakeProcessingStep()
+
+
+@pytest.fixture
+def processing_step() -> ProcessingStrategy[KafkaPayload]:
+    return FakeProcessingStep()
+
+
+@pytest.fixture
+def valid_message() -> Message[KafkaPayload]:
+    valid_payload = KafkaPayload(b"", b"", [])
+    return Message(Partition(Topic(""), 0), 0, valid_payload, datetime.now())
+
+
+@pytest.fixture
+def invalid_message() -> Message[KafkaPayload]:
+    invalid_payload = KafkaPayload(None, b"", [])
+    return Message(Partition(Topic(""), 0), 0, invalid_payload, datetime.now())
+
+
+def test_raise(
+    valid_message: Message[KafkaPayload],
+    invalid_message: Message[KafkaPayload],
+    processing_step: FakeProcessingStep,
+) -> None:
+    dlq_raise = DeadLetterQueue(processing_step, RaiseInvalidMessagePolicy())  # type: ignore
```

Comment:
How come `type: ignore` was needed? Can we fix instead?

Reply:
I keep getting a `Cannot infer type argument 1 of "DeadLetterQueue"` mypy error, I'm not sure why. Is there something up with how I've defined the Fake step/policy classes?

Reply:
Why don't you specify the type?
```
dlq_raise: DeadLetterQueue[KafkaPayload] = DeadLetterQueue(processing_step, RaiseInvalidMessagePolicy())
```

Reply:
Ah ok thanks 🤦‍♂️